test/ComparedQueryable.Test/ComparedQueryableTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Xunit;
+
+namespace ComparedQueryable.Test
+{
+    public class ComparedQueryableTest

```

Comment:
This is a good suggestion, although I don't know if there's a clear answer for what the desired behavior should be. The first thing that came to mind for me was similar to yours: "123 N 2nd St" vs "123 N 10th St". But what about "S/N 5EFFH" vs "S/N 15AAA"? I would want 15AAA to come before 5EFFH. So perhaps it's a word boundary that defines the appropriate sorting behavior? Maybe we only sort numbers that come at the beginning of the string? Although I don't think that's going to be satisfactory (consider "Unit 1", etc.)
But: additional tests will help document what we define as "correct" behavior in these scenarios.